### ISSUE TITLE
[AC] [Crash] BG Bots Crash Server

### ISSUE THREAD

**Issue by whipowill** [trickerer/Trinity-Bots#941]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that my issue may get **closed without notice** if I *intentionally* remove or skip any mandatory\* field

### Current behaviour

I upgraded my server from an older version of this NPCBots repo to use your latest version.  Bc I was bringing over my old ``acore_characters`` table, I had to manually run your MySQL files as instructed in the install directions.  

The server works fine and everything is in order.  I am able to manually spawn bots and the wandering bots work fine.  The issue is when I create a battleground the server crashes. My config has this activated so it should work:

```
NpcBot.WanderingBots.BG.Enable = 1
```

### Expected behaviour

I'm expecting battleground bots to enter the game as they used to in previous versions of this repo, without immediately crashing the server.

### Crashlog

No logs generated

### Steps to reproduce the problem

Install this repo, turn on bots in battlegrounds in the config, launch server, enter battleground, immediate crash.

### TrinityCore or AzerothCore

AzerothCore

### Core rev. hash/commit

dc9d4972c3fb+

### Operating system

Ubuntu 24

**Comment by trickerer** [comment]
Post a debug crashlog

**Comment by whipowill** [comment]
Forgive me if it's a skill issue, but there is no error log generated.

**Comment by trickerer** [comment]
It isn't crashing for me so without a crashlog I won't be able to help you.
https://github.com/trickerer/Trinity-Bots/issues/650#issuecomment-1751915116

**Comment by Ramvaris** [comment]
I got the same problem.

I just started the worldserver in debug mode with the gdb log for you.
I attached the crash file.
Just the username for the root folder has been replaced for security reasons but that shouldn't be relevant for the debugging.

The problem happens whenever a BG starts. Be it Warsong, Arathi or Alterac. It all worked before the current update cycle (every monday for me).

[gdb.txt](https://github.com/user-attachments/files/17978186/gdb.txt)

**Comment by trickerer** [comment]
https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots/commit/cde40813c0fae7ce8541a8a680db1959cce75ab6
https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots/commit/f588a094d40fd0e08c911e8b8fc3216cd10f5d77

**Comment by Shadowcoldzx** [comment]
I had the same problem and the second commit is how I fixed it originally (well I actually just commented out the whole assert line). After seeing this and adding back the if condition I see no changes, everything still works just fine. The error log was pointing to the assert on every crash in my case.
Since I've never done anything in C I'm more than happy to see we came to the same problem line :>
